### PR TITLE
Fix "Covered through ___" date

### PR DESCRIPTION
**Pull request by forevermatt**
## Fixed
- Fix `formatFriendlyDate()` to show the right date regardless of timezone

---

Previously, a date/time string of "2021-12-31" was showing as "12/30/2021" due to timezone issues.

### CODE REVIEW

**Review by hobbitronics — CHANGES_REQUESTED** [review]
can you look at `components/dates.ts` and see if that does the same thing and we should probably combine these files anyways. the formatDate function uses a different method to deal with the utc/local trap, but not sure if these are serving different purposes.

File: src/helpers/date.ts [hunk 1]
```diff
@@ -1,4 +1,14 @@
+const iso8601dateRegex = /[0-9]{4}-[0-9]{2}-[0-9]{2}/
+
+const isJustDate = (dateTimeString: string) => iso8601dateRegex.test(dateTimeString)
+
 export const formatFriendlyDate = (dateTimeString: string) => {
+  if (isJustDate(dateTimeString)) {
```
Comment (RussellChamp) [inline src/helpers/date.ts:6]: we've already included date-fns in the project. does this do the same thing that parseISO would? https://date-fns.org/v2.27.0/docs/parseISO

**Comment by forevermatt** [comment]
@RussellChamp or @hobbitronics - Is one of you working on resolving/fixing this? As I'm cleaning up old PRs, I saw this one I'd essentially abandoned still sitting here, and I see that the date is still incorrect on production.

**Comment by hobbitronics** [comment]
> @RussellChamp or @hobbitronics - Is one of you working on resolving/fixing this? As I'm cleaning up old PRs, I saw this one I'd essentially abandoned still sitting here, and I see that the date is still incorrect on production.

No, I haven't been working on it.

File: src/components/FilePreview/FilePreview.svelte [hunk 2]
```diff
@@ -1,10 +1,10 @@
 <script lang="ts">
 import Banner from '../components/Banner.svelte'
 import type { ClaimFile } from 'data/claims'
-import { formatDate } from '../dates'
+import { formatDate } from '../../helpers/dates'
 import { createEventDispatcher } from 'svelte'
 import { flip } from 'svelte/animate'
-import { Button, Progress } from '@silintl/ui-components'
```
Comment (hobbitronics) [inline src/components/FilePreview/FilePreview.svelte:7]: Button was unused

File: src/components/ClaimsTable.svelte [hunk 3]
```diff
@@ -32,7 +32,7 @@ export let title = ''
             <a href={customerClaimDetails(policyId, claim.id)}>{claim.reference_number || ''}</a>
             ({claim.status})
           </Datatable.Data.Row.Item>
-          <Datatable.Data.Row.Item>{formatFriendlyDate(claim.incident_date)}</Datatable.Data.Row.Item>
+          <Datatable.Data.Row.Item>{getJustDateFromUTCDateString(claim.incident_date)}</Datatable.Data.Row.Item>
```
Comment (hobbitronics) [inline src/components/ClaimsTable.svelte:35]: was showing 'Invalid date'

File: src/components/forms/ClaimForm.svelte [hunk 4]
```diff
@@ -157,9 +157,10 @@ const onSaveForLater = (event: Event) => {
 }
 
 const getFormData = () => {
+  const date = lostDate === todayDateString.split('T')[0] ? todayDateString : lostDate
```
Comment (hobbitronics) [inline src/components/forms/ClaimForm.svelte:160]: preserve the full ISOString if today

**Review by hobbitronics — APPROVED** [review]
I approve

File: src/customTypes/@silintl/ui-components.d.ts [hunk 5]
```diff
@@ -144,7 +144,7 @@ declare module '@silintl/ui-components' {
 
   interface MoneyInputProps extends svelte.JSX.HTMLAttributes<HTMLElementTagNameMap['div']> {
     label?: string
-    value?: string
+    value?: string | number
```
Comment (Schparky) [inline src/customTypes/@silintl/ui-components.d.ts:147]: Can this just be `number`?

**Review by Schparky — APPROVED** [review]

File: src/customTypes/@silintl/ui-components.d.ts (hunk 5 above)
Comment (hobbitronics) [reply src/customTypes/@silintl/ui-components.d.ts:147]: I think so, no new errors if I do.

**Pull request merged**